### PR TITLE
Fix Typos and Add Header Links to RAG Demo

### PR DESCRIPTION
**Pull request by kodyfisher**
Fix typos, some grammatical changes and add linkable headers to [full RAG Demo ](https://docs.tecton.ai/docs/tutorials/full_rag)

### CODE REVIEW

File: restaurant-recommender/full-rag-demo.ipynb [hunk 1]
```diff
@@ -27,7 +27,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Install Packages"
+    "## Install Packages {#install-packages}"
```
Comment (brian-tecton-ai) [inline restaurant-recommender/full-rag-demo.ipynb]: Why is this necessary? The headers are already linkable:
https://docs.tecton.ai/docs/tutorials/full_rag#install-packages

Comment (kodyfisher) [reply restaurant-recommender/full-rag-demo.ipynb]: They are not all linkable. Example Support Chat Bot, Simple Prompt, Features as Tools, Knowledge are not, for example

You are correct that some of these are already linkable, though. I will look into fixing the PR

Comment (brian-tecton-ai) [reply restaurant-recommender/full-rag-demo.ipynb]: Okay I think it's because of the header number. `# Example Support Chat Bot` vs `## Install Packages`. Likely this is because Docusaurus expects the single # to indicate the title of the page. If we update all the headers here to have an extra `#`, I think we should be good to go 👍

File: restaurant-recommender/full-rag-demo.ipynb [hunk 2]
```diff
@@ -370,22 +371,22 @@
     "Sure, Jim! Could you please share your preferences, such as the type of cuisine you enjoy or any specific dietary  \n",
     "restrictions? That way, I can recommend a restaurant that you'll love.\n",
     "```\n",
-    "This makes a lot of sense because it the user's food preference is not part of the *prompt*.\n",
+    "This makes a lot of sense because the user's food preference is not part of the *prompt*.\n",
     "\n",
-    "Adding food preference to the prompt would be a good option and you can test that out on your own. But there is an alternative that we explore in the next section, Features as Tools."
+    "Adding food preference to the prompt would be a good option which you can test that out on your own. But there is an alternative that we explore in the next section, Features as Tools."
```
Comment (sergioferragut) [inline restaurant-recommender/full-rag-demo.ipynb]: which you can test that out on your own. << remove "that"

File: restaurant-recommender/full-rag-demo.ipynb [hunk 3]
```diff
@@ -751,13 +751,13 @@
...
     "\n",
-    "The following cell creates a sample *knowledge* base for the LLM using `make_local_source`, notice that the description `\"Restaurants that signed up for the service\"` provides the LLM the information it needs to follow the new instruction in the prompt created above. \n",
+    "The following cell creates a sample *knowledge* base for the LLM using `make_local_source`. Notice that the description `\"Restaurants that signed up for the service\"` provides the LLM the information it needs to follow the new instruction in the prompt created above. \n",
     "\n",
     "The `source_as_knowledge` function that creates `restaurant_knowledge` calculates embedding vectors and loads the data into a vector search database.\n",
     "\n",
-    "The complete *AgentService* will now contains the prompt, two features as tools and the newly created `restaurant_knowledge` with the intent that it only recommend restaurants that have signed up for the service."
+    "The complete *AgentService* will now contains the prompt, two features as tools and the newly created `restaurant_knowledge` with the intent that to only recommend restaurants that have signed up for the service."
```
Comment (sergioferragut) [inline restaurant-recommender/full-rag-demo.ipynb]: intent that to -> remove "that"

**Review by sergioferragut — APPROVED** [review]
Thanks for the thorough grammar/typo review. Just caught a couple more in my pass. 
Other than that, the changes are great.

File: restaurant-recommender/full-rag-demo.ipynb (hunk 1 above)
Comment (kodyfisher) [reply restaurant-recommender/full-rag-demo.ipynb]: Good catch! Fixed
